webpack/components/EditableTextInput/EditableTextInput.js

```diff
@@ -0,0 +1,98 @@
+import React, { useState, Fragment } from 'react';
+import { TextInput, TextArea, Text, TextVariants, Button, Split, SplitItem } from '@patternfly/react-core';
+import { TimesIcon, CheckIcon, EditIcon } from '@patternfly/react-icons';
+import PropTypes from 'prop-types';
+import Loading from '../Loading';
+import './editableTextInput.scss';
+
+const EditableTextInput = ({
+  onEdit, value, textArea, editable, label,
+}) => {
+  // Tracks input box state
+  const [inputValue, setInputValue] = useState(value);
+  const [editing, setEditing] = useState(false);
+  const [working, setWorking] = useState(false);
+  const onSubmit = () => {
+    setWorking(true);
+    onEdit(inputValue);
+    setWorking(false);
+    setEditing(false);
+  };
+  const onClear = () => {
+    setInputValue(value);
+    setEditing(false);
+  };
+
+  const textInput = () => {
+    const sharedProps = {
+      value: inputValue,
+      onChange: v => setInputValue(v),
+    };
+
+    if (textArea) {
+      return <TextArea {...sharedProps} aria-label={`text area ${label}`} />;
+    }
+    return <TextInput {...sharedProps} type="text" aria-label={`text input ${label}`} />;
+  };
```

Comment:
I don't think this const is necessary.  Why not just put everything inline in JSX?
```javascript
{
  textArea && <TextArea value={inputValue} onChange={v => setInputValue(v)} aria-label={`text area ${label}` />
}
{ 
  !textArea && <TextInput value={inputValue} onChange={v => setInputValue(v)} aria-label={`text input ${label}`} />
}
```
Having more of the markup in JSX like this lets me get a clearer picture in my head of the structure of the page, and saves me from having to jump around the code when reading it.

Reply:
I moved it into JSX but as a ternary for the reasons mentioned in the other comment

Reply:
👿 

Actually this is fine, it's not too long.. :smile: